File: backend/pokerogue/src/main/java/com/pokerogue/helper/type/dto/PokemonTypeResponse.java
package com.pokerogue.helper.type.dto;

import com.pokerogue.helper.type.data.Type;

public record PokemonTypeResponse(String typeName, String typeLogo) {

    public static PokemonTypeResponse from(Type type) {
        return new PokemonTypeResponse(type.getKoName(), type.getImage());
    }
}

File: android/app/src/main/java/poke/rogue/helper/remote/dto/response/type/PokemonTypeResponse.kt
package poke.rogue.helper.remote.dto.response.type

import kotlinx.serialization.Serializable
import poke.rogue.helper.data.model.Type
import java.util.Locale

@Serializable
data class PokemonTypeResponse(
    val pokemonTypeName: String,
    val pokemonTypeLogo: String,
)

fun PokemonTypeResponse.toData(): Type = Type.valueOf(pokemonTypeName.uppercase(Locale.ROOT))

fun List<PokemonTypeResponse>.toData(): List<Type> = map(PokemonTypeResponse::toData)
